signer/src/schnorr.rs

```diff
@@ -0,0 +1,156 @@
+//! Mina Schnorr signature scheme
+//!
+//! An implementation of the singer interface for the Mina signature algorithm
+//!
+//! Details: <https://github.com/MinaProtocol/mina/blob/develop/docs/specs/signatures/description.md>
+
+use ark_ec::{
+    AffineCurve,     // for prime_subgroup_generator()
+    ProjectiveCurve, // for into_affine()
+};
+use ark_ff::{
+    BigInteger, // for is_even()
+    Field,      // for from_random_bytes()
+    PrimeField, // for from_repr()
+    Zero,
+};
+use blake2::{
+    digest::{Update, VariableOutput},
+    VarBlake2b,
+};
+use oracle::{
+    poseidon::{SpongeConstants, SpongeState},
+    rndoracle::{ArithmeticSponge, Sponge},
+};
+use std::ops::Neg;
+
+use crate::{
+    BaseField, CurvePoint, FieldHelpers, Hashable, Keypair, NetworkId, PubKey, ROInput,
+    ScalarField, Signable, Signature, Signer,
+};
+
+/// Schnorr signer context for the Mina signature algorithm
+///
+/// For details about the signature algorithm please see [crate::schnorr]
+pub struct Schnorr<SC: SpongeConstants> {
```

Comment:
`struct SignerCtx` perhaps?

Reply:
Schnorr implements Signer interface, so it can be called whatever it wants as long as it implements the Signer interface.  I thought this is nice, so we can at this level just call things what they are under the hood, which isn't really visible to the user of the API